codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -416,6 +416,9 @@ private static HeaderType validate(int streamId, CharSequence name,
             throw streamError(streamId, PROTOCOL_ERROR,
                     "Illegal value specified for the 'TE' header (only 'trailers' is allowed).");
         }
+        if (!isValidHeaderValue(value)) {
+            throw streamError(streamId, PROTOCOL_ERROR, "Illegal header value given for header '%s'.", name);
```

Comment:
The header value can contain sensitive info, we can not include the whole value in the exception message. But without any hint, it's harder to debug what's wrong with the value. One of the most frequent case in my practice is to have `\n` at the end of the value. 
Consider including the hex value of the invalid byte (and maybe event its position) in the exception message. Lots of users will say "thank you" if they don't have to go through frame logging in order to infer what is wrong with the header value.